### ISSUE TITLE
csvstorage: files in source are rewritten

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#987]
csvstorage modifies its config files and tests fail:

```
Some tests report warnings because of invalid CSV files
/home/markus/Projekte/Elektra/current/src/plugins/csvstorage/testmod_csvstorage.c:61: error in testreadwriteinvalid: no warnings in kdbGet
/home/markus/Projekte/Elektra/current/src/plugins/csvstorage/testmod_csvstorage.c:62: error in testreadwriteinvalid: error: wrote invalid data
/home/markus/Projekte/Elektra/current/src/plugins/csvstorage/testmod_csvstorage.c:79: error in testwriteinvalidheader: no warnings in kdbGet
/home/markus/Projekte/Elektra/current/src/plugins/csvstorage/testmod_csvstorage.c:80: error in testwriteinvalidheader: error: wrote invalid data
```

Please also try to enable ENABLE_DEBUG and fix problems occurring there.

**Comment by markus2330** [comment]
seems to be a problem because of crashes with ENABLE_DEBUG.

Is it possible that the tests write into the source? They should only write into the build directory (passed as parameter), as other similar test cases do.

**Comment by tom-wa** [comment]
can't reproduce it build with
```
 % kdb export system/info/constants/cmake
= All cmake variables as defined in the file cmake/ElektraCache.cmake
BINDINGS = cpp;intercept
BUILD_FULL = OFF
BUILD_SHARED = ON
BUILD_STATIC = OFF
BUILTIN_DATA_FOLDER = /usr/local/share/elektra/test_data
BUILTIN_EXEC_FOLDER = /usr/local/lib/elektra/tool_exec
BUILTIN_PLUGIN_FOLDER = /usr/local/lib/elektra
CMAKE_INSTALL_PREFIX = /usr/local
ENABLE_DEBUG = ON
ENABLE_LOGGER = OFF
GTEST_ROOT =
KDB_DB_DIR = .dir
KDB_DB_FILE = default.ini
KDB_DB_HOME = /home
KDB_DB_INIT = elektra.ini
KDB_DB_SPEC = share/elektra/specification
KDB_DB_SYSTEM = /etc/kdb
KDB_DB_USER = .config
KDB_DEFAULT_RESOLVER = resolver_fm_hpu_b
KDB_DEFAULT_STORAGE = ini
LIB_SUFFIX =
PLUGINS = base64;blockresolver;boolean;c;cachefilter;ccode;conditionals;constants;counter;crypto_botan;crypto_gcrypt;crypto_openssl;csvstorage;curlget;dbus;desktop;doc;dpkg;dump;enum;error;errtest;filecheck;fstab;gitresolver;glob;hexcode;hidden;hosts;iconv;ini;iterate;jni;journald;keytometa;line;lineendings;list;logchange;lua;mathcheck;mozprefs;network;ni;noresolver;null;passwd;path;profile;python;python2;regexstore;rename;required;resolver_fm_b_b;resolver_fm_hb_b;resolver_fm_hp_b;resolver_fm_hpu_b;resolver_fm_pb_b;resolver_fm_ub_x;resolver_fm_uhb_xb;resolver_fm_xb_x;resolver_fm_xhp_x;resolver_fm_xp_x;semlock;shell;simpleini;simplespeclang;spec;struct;sync;syslog;tcl;template;timeofday;tracer;type;uname;validation;wresolver;xmltool;yajl
TARGET_CMAKE_FOLDER = lib/cmake/Elektra
TARGET_INCLUDE_FOLDER = elektra
TARGET_PKGCONFIG_FOLDER = pkgconfig
TARGET_PLUGIN_FOLDER = elektra
TARGET_TEMPLATE_FOLDER = share/elektra/templates
TARGET_TEST_DATA_FOLDER = share/elektra/test_data
TARGET_TOOL_EXEC_FOLDER = lib/elektra/tool_exec
TOOLS = kdb
```

**Comment by markus2330** [comment]
To reproduce:

```
ls -l atime src/plugins/csvstorage/csvstorage
-rw-r--r-- 1 markus users 25 Oct 16 00:39 invalid_columns.csv
-rw-r--r-- 1 markus users 55 Oct 16 00:39 invalid_columns_header2.csv
# run unit tests with make run_all
ls -l atime src/plugins/csvstorage/csvstorage
-rw-r--r-- 1 markus users 25 Oct 16 09:13 invalid_columns.csv
-rw-r--r-- 1 markus users 55 Oct 16 09:13 invalid_columns_header2.csv
```

Expected behaviour: the files should be untouched. (use build directory to write files).

**Comment by tom-wa** [comment]
should be fixed with #1027
